src/java.desktop/macosx/native/libawt_lwawt/awt/AWTWindow.m

```diff
@@ -1677,19 +1677,17 @@ + (AWTWindow *) lastKeyWindow {
             NSRect screenRect = [[nsWindow screen] frame];
             [nsWindow setFrame:screenRect display:YES];
         } else {
-            [JNFException raise:[ThreadUtilities getJNIEnv]
-                             as:kRuntimeException
-                         reason:"Failed to enter full screen."];
+            [NSException raise:@"Java Exception" reason:@"Failed to enter full screen." userInfo:nil];
```

Comment:
Since this code is lexically inside a block executed on the AppKit thread it makes no sense to raise a Java Exception.